contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -1260,14 +1306,19 @@ BEGIN
 		day_diff = sys.datepart('day', enddate OPERATOR(sys.-) startdate);
 		result = day_diff;
 	WHEN 'day' THEN
-		day_diff = sys.datepart('day', enddate OPERATOR(sys.-) startdate);
-		result = day_diff;
+		y1 = sys.datepart('year', enddate);
+		m1 = sys.datepart('month', enddate);
+		d1 = sys.datepart('day', enddate);
+		y2 = sys.datepart('year', startdate);
+		m2 = sys.datepart('month', startdate);
+		d2 = sys.datepart('day', startdate);
+		result = sys.daydiff_between_dates(d1, m1, y1, d2, m2, y2);
 	WHEN 'week' THEN
 		day_diff = sys.datepart('day', enddate OPERATOR(sys.-) startdate);
 		result = day_diff / 7;
 	WHEN 'hour' THEN
 		day_diff = sys.datepart('day', enddate OPERATOR(sys.-) startdate);
```

Comment:
is it guaranteed that start date is smaller than enddate?

Reply:
We should surely have negative and boundary checks. There are no guarantees.

Reply:
how will this work for say 2 :1: 00 - 1:23:00, so it will be 1 (day_diff) * 24 + 22 (abs hour diff), wouldn't negative value be correct